main.nf

```diff
@@ -116,6 +116,13 @@ workflow.onComplete {
     log.info "Execution duration: $workflow.duration"
 }
 
+if( (!nextflow.version.matches('>=19.04.2'))) {
+    error "This workflow requires Nextflow (>=19.04.2, <=21.12.1) -- You are running version $nextflow.version"
+}
+if( (nextflow.version.matches('>21.12.1.edge'))) {
```

Comment:
I don't think we'll ever hit this condition, turns out versions of Nextflow that don't support DSL1 error out before running any part of the pipeline script.

You can keep it for safe keeping (also works as a pseudo docstring of the versions we support), or transform it in a comment.